src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -438,6 +438,37 @@ JNI_METHOD(void, setDeviceAttestationDelegate)
     }
 }
 
+JNI_METHOD(void, setAttestationTrustStoreDelegate)
+(JNIEnv * env, jobject self, jlong handle, jobject attestationTrustStoreDelegate)
+{
+    chip::DeviceLayer::StackLock lock;
+    CHIP_ERROR err                           = CHIP_NO_ERROR;
+    AndroidDeviceControllerWrapper * wrapper = AndroidDeviceControllerWrapper::FromJNIHandle(handle);
+
+    ChipLogProgress(Controller, "setAttestationTrustStoreDelegate() called");
+
+    if (attestationTrustStoreDelegate != nullptr)
+    {
+        wrapper->ClearAttestationTrustStoreBridge();
+        jobject attestationTrustStoreDelegateRef = env->NewGlobalRef(attestationTrustStoreDelegate);
+        wrapper->SetAttestationTrustStoreBridge(new AttestationTrustStoreBridge(attestationTrustStoreDelegateRef));
+        VerifyOrExit(wrapper->GetAttestationTrustStoreBridge() != nullptr, err = CHIP_ERROR_NO_MEMORY);
+
+        wrapper->ClearDeviceAttestationVerifier();
+        wrapper->SetDeviceAttestationVerifier(new Credentials::DefaultDACVerifier(wrapper->GetAttestationTrustStoreBridge()));
```

Comment:
Why new a DefaultDACVerifier, not re-initialize it with
GetDefaultDACVerifier(wrapper->GetAttestationTrustStoreBridge())?

Reply:
`GetDefaultDACVerifier(wrapper->GetAttestationTrustStoreBridge()) ` cloud not re-initialize the DefaultDACVerifier.

```
DeviceAttestationVerifier * GetDefaultDACVerifier(const AttestationTrustStore * paaRootStore)
{
    static DefaultDACVerifier defaultDACVerifier{ paaRootStore };

    return &defaultDACVerifier;
}
```

Static local variables will only be initialized when called first time, so the `paaRootStore` cannot be replaced.